slider.js

```diff
@@ -30,115 +46,97 @@ const createSlider = (function () {
     };
   };
 
-  const getPlaceholder = (sliderRoot, customPlaceholder) => {
-    const imageContainer = document.createElement("div");
-    imageContainer.classList.add("placeholder");
-    const image = document.createElement("img");
-    const imageParent = sliderRoot.querySelector(
-      SLIDER_CLASSES.SLIDER_CONTAINER
-    );
-    image.src = customPlaceholder;
-    image.alt = "Placeholder image";
-    imageContainer.appendChild(image);
-    imageParent.appendChild(imageContainer);
-
-    sliderRoot.querySelector(SLIDER_CLASSES.NAVIGATION_BUTTONS).remove();
+  const createSlide = (src, alt) => {
+    const slide = document.createElement("li");
+    slide.classList.add(".slider-item");
+    slide.innerHTML = `<img
+        width="400"
+        height="200"
+        class="image"
+        src="${src}"
+        alt="${alt}"
+      />`;
+
+    return slide;
   };
 
-  const getPrevSlideIndex = (activeSlideIndex, slides) => {
+  const getPrevSlideIndex = () => {
     return activeSlideIndex === 0 ? slides.length - 1 : activeSlideIndex - 1;
   };
 
-  const getNextSlideIndex = (activeSlideIndex, slides) => {
+  const getNextSlideIndex = () => {
+    console.log();
     return activeSlideIndex >= slides.length - 1 ? 0 : activeSlideIndex + 1;
   };
 
-  const selectSlide = (slides, newSlideIndex) => {
+  const selectSlide = (newSlideIndex) => {
     slides.forEach((slide) =>
       slide.classList.remove(SLIDER_CLASSES.ITEM_ACTIVE)
     );
     slides[newSlideIndex].classList.add(SLIDER_CLASSES.ITEM_ACTIVE);
+    activeSlideIndex = newSlideIndex;
   };
 
-  const getPrevSlide = (slides, activeSlideIndex) => {
-    const prevSlideIndex = getPrevSlideIndex(activeSlideIndex, slides);
-    selectSlide(slides, prevSlideIndex);
-
-    return prevSlideIndex;
-  };
-
-  const getNextSlide = (slides, activeSlideIndex) => {
-    const nextSlideIndex = getNextSlideIndex(activeSlideIndex, slides);
-    selectSlide(slides, nextSlideIndex);
-
-    return nextSlideIndex;
+  const selectPrevSlide = () => {
+    const prevSlideIndex = getPrevSlideIndex();
+    selectSlide(prevSlideIndex);
+    initAutoscroll();
```

Comment:
I think initAutoscroll(); shouldn’t be passed here. This applies only for selectNextSlide().

Reply:
I was thinking about avoiding it here. Code is not beautiful because of it, however it makes autoscroll more nice:

when a user clicks navigation, autoscroll timer is reset. If we don't do it, user clicks navigation and autoscroll timer calls nextSlide every 3 second. It becomes glitchy